### PR TITLE
Automate test suite docs

### PR DESCRIPTION
**Pull request by heyste**
Testing the future GHA workflow for generating [Conformance test suite documentation](https://github.com/cncf/k8s-conformance/tree/master/docs)

### CODE REVIEW

File: docs/check-conformance-test-suite-docs.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+function setup_k8s_repo() {
+  mkdir -p /tmp/go/src/k8s.io
+  cd "$_"
+  git clone --depth 1 --branch "${RELEASE_VERSION}" https://github.com/kubernetes/kubernetes.git
+  export KUBE_ROOT=/tmp/go/src/k8s.io/kubernetes
+  cd "${KUBE_ROOT}"
+  git log -1
+}
+
+function install_go() {
```
Comment (BobyMCbobs) [inline docs/check-conformance-test-suite-docs.sh:46]: @heyste, if this is run in a development environment with an existing Go installation, will it affect the existing installation?

File: docs/check-conformance-test-suite-docs.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  cd "${KUBE_ROOT}"
+  ./_output/bin/ginkgo --dry-run=true --focus='[Conformance]' ./_output/bin/e2e.test -- --spec-dump "${KUBE_ROOT}/_output/specsummaries.json" > /dev/null
+
+  DOC_VERSION="$(echo "$STABLE_VERSION" | awk -F '.' '{print $1 "." $2}' | sed 's/v//')"
+  go run ./test/conformance/walk.go --version="$DOC_VERSION" \
+                                  --url https://github.com/kubernetes/kubernetes/tree/release-"${DOC_VERSION}"/ \
+                                  --docs ./_output/specsummaries.json > ./_output/KubeConformance-"${DOC_VERSION}".md
+  sed "s|tree/release-${DOC_VERSION}.*/test/e2e|tree/release-${DOC_VERSION}/test/e2e|g" ./_output/KubeConformance-"${DOC_VERSION}".md > "${GITHUB_WORKSPACE}"/docs/KubeConformance-"${DOC_VERSION}".md
+  echo "${RELEASE_VERSION}" > /tmp/release-version
+  echo "Conformance test suite documention for $(cat /tmp/release-version) has been created"
```
Comment (BobyMCbobs) [inline docs/check-conformance-test-suite-docs.sh]: suggestion: change `$(cat /tmp/release-version)` to `$(< /tmp/release-version)`

File: docs/check-conformance-test-suite-docs.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+function main() {
+  echo "Executing script: ${GITHUB_WORKSPACE}/docs/${BASH_SOURCE[0]}"
```
Comment (BobyMCbobs) [inline docs/check-conformance-test-suite-docs.sh]: Getting the error
```
./docs/check-conformance-test-suite-docs.sh: line 70: GITHUB_WORKSPACE: unbound variable
```

you can set this to `${GITHUB_WORKSPACE:-}` to default it to nothing if it's undeclared

**Comment by BobyMCbobs** [comment]
Getting the output of `Conformance test suite documentation for v1.25.0 was found. No further action required`.
It appears to be checking against the cncf/k8s-conformance repo.
I've swapped it out for this repo in the code and will see the response

**Comment by heyste** [comment]
`export CONFORMANCE_BASE_URL=https://raw.githubusercontent.com/ii/automate-conformance-docs/master/docs` will test again the current repo which is missing the current 1.25 document.

File: docs/check-conformance-test-suite-docs.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  RESPONSE_CODE=$(curl -s -o /dev/null "$MD_FILE_URL" --write-out "%{http_code}")
+  if [[ $RESPONSE_CODE ==  "404" ]]; then
+    TEST_SUITE_DOCUMENT="missing"
+  fi
+}
+
+function setup_k8s_repo() {
+  mkdir -p /tmp/go/src/k8s.io
+  cd "$_"
+  git clone --depth 1 --branch "${RELEASE_VERSION}" https://github.com/kubernetes/kubernetes.git
```
Comment (BobyMCbobs) [inline docs/check-conformance-test-suite-docs.sh]: suggestion:
```
if [ ! -d /tmp/go/src/k8s.io/kubernetes ]; then
  git clone --depth 1 --branch "${RELEASE_VERSION}" https://github.com/kubernetes/kubernetes.git
fi
```

File: docs/check-conformance-test-suite-docs.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  export KUBE_ROOT=/tmp/go/src/k8s.io/kubernetes
+  cd "${KUBE_ROOT}"
+  git log -1
+}
+
+function install_go() {
+  cd ${KUBE_ROOT}
+  GO_VERSION=$(grep -A1 "golang: upstream version" build/dependencies.yaml | tail -1 | awk -F ':' '{print $2}' | sed 's/^ *//')
+  echo "GO_VERSION: $GO_VERSION"
```
Comment (BobyMCbobs) [inline docs/check-conformance-test-suite-docs.sh:49]: suggestion: don't install Go, if the exact same version is already installed

File: docs/check-conformance-test-suite-docs.sh (hunk 3 above)
Comment (heyste) [reply docs/check-conformance-test-suite-docs.sh]: Fixed

File: docs/check-conformance-test-suite-docs.sh (hunk 5 above)
Comment (heyste) [reply docs/check-conformance-test-suite-docs.sh:49]: Fixed

**Comment by BobyMCbobs** [comment]
lgtm!

**Pull request merged**